version: '3.9'

services:
  postgres:
    image: postgres:17.6-alpine3.22
    container_name: cti_postgres
    restart: unless-stopped
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    networks:
      - app-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  # backend:
  #   build:
  #     context: ./backend
  #     dockerfile: Dockerfile.dev
  #   container_name: cti_backend
  #   env_file:
  #     - .env
  #   depends_on:
  #     postgres: 
  #       condition: service_healthy
  #   ports:
  #     - "3000:3000"
  #   volumes:
  #     - ./backend:/app
  #     - /app/node_modules
  #   networks:
  #     - app-network

volumes:
  postgres_data:
  
networks:
  app-network: 
    driver: bridge
